university/src/university/University.java

```diff
@@ -0,0 +1,62 @@
+package university;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class University {
+    private static University UNIVERSITYINSTANCE;
```

Comment:
I am probably nit-picking right now, but `UNIVERSITY_INSTANCE` looks slightly better. Words should be separated with underscore (`_`) characters